Source/WebCore/platform/cocoa/VideoPresentationModelVideoElement.mm

```diff
@@ -215,13 +215,13 @@
         m_videoElement->setVideoFullscreenFrame(rect);
 }
 
-void VideoPresentationModelVideoElement::setVideoSizeFenced(const FloatSize& size, WTF::MachSendRight&& fence)
+void VideoPresentationModelVideoElement::setVideoSizeFenced(const FloatSize& size, WTF::MachSendRight&& fence, CompletionHandler<void()>&& completionHandler)
 {
     if (!m_videoElement)
         return;
 
     INFO_LOG_IF_POSSIBLE(LOGIDENTIFIER, size);
-    m_videoElement->setVideoLayerSizeFenced(size, WTFMove(fence));
+    m_videoElement->setVideoLayerSizeFenced(size, WTFMove(fence), WTFMove(completionHandler));
     m_videoElement->setVideoFullscreenFrame({ { }, size });
```

Comment:
It seems weird to me that we would pass a completion handler to `setVideoLayerSizeFenced`, but then continue doing more work in this method (especially if `setVideoFullscreenFrame` has side-effects.

Reply:
That is a good point. I have addressed this in the latest patch.

Thanks for reviewing!